docs/PERSISTENCE.md

```diff
@@ -0,0 +1,57 @@
+- Read: Since we only use 3 Plog Servers to provide redundancy, we are imagining the Plogs with the same Plog Id in different Plog Servers are exactly the same. So when the client sends a read request, it will only send to one Plog Server in the Persistence Cluster. The Plog Server valid the reading request and return the corresponding response. The I/O size for each append operation has a upper limit, currently is set to be one plog size, which is 16MB.
+- Seal: The client will send the seal request to the Plog Servers with a sealed offset. If this offset is inconsistent with the current offset the Plog Server holds, it will seal this Plog with the smaller offset. Once a Plog is Sealed, the client will not be allowed to make any modifications. 
+
+## Append Request Reassembly （TODO）
+It is possible that the arriving append request will arrive at the Plog Server out of order due to bad networking conditions. Therefore, on the Plog Server-side, we should cache the out-of-order requests. Once the Plog Server receives an append request, it will first check whether the request's offset is equal to the current offset of the request Plog it holds. If they are the same, this is an in-order request, we can handle it with the normal logic. Otherwise, it is an out-of-order request, the Plog Server will first reply to the Plog Client, notify the client that this packet is out-of-order. Then, it will cache this request. The append request cache is a linked list. For each out-of-order request, we insert it into the cache, sorting with its offset. Then, the Plog Server checked the neighboring requests of the inserted request, checking whether the neighboring append requests can merge. We can merge the append requests with continuous offset. If the Plog Server receives the missing append request, it will process the requests in the cache as well. 
+
+# Log Stream
+The log stream is an interface for clients to continuously read and append data from/to multiple plogs without worrying to handle the operations of plogs. Conceptually it is a list of plogs that only last one are unsealed and can append data. It could support different types of data services such as WAL, Indexer snapshot, etc. The log stream will provide API to read/append the data, store all the metadata information such as used plog id and their offsets. In order to handle the case that a log stream crash, we need to record these metadata as well. We store the metadata of all data services in the same partition into `Metadata Manager`, which we will describe later.
```

Comment:
```suggestion
The LogStream is a K2 client library which provides a similar interface to PlogClient (create/read/append), but without any size constraints.  Users simply create a LogStream, and can append as much (or as big) data as they need. Random reads over the entire stream are also supported. The library solves the problem of managing multiple PLOG blocks to form a single, continuous, infinite(2^64) stream. Conceptually it is a list of plogs where only the last one is unsealed and can append data. It could support different types of data services such as WAL, Indexer snapshot, etc. The log stream will provide API to read/append the data, store all the metadata information such as used plog id and their offsets. In order to handle the case that a log stream crash, we need to record these metadata as well. We store the metadata of all data services in the same partition into `Metadata Manager`, which we will describe later.
```